packages/@aws-cdk/aws-apigatewayv2/lib/http/vpc-link.ts

```diff
@@ -0,0 +1,109 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import { IResource, Lazy, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnVpcLink } from '../apigatewayv2.generated';
+
+/**
+ * Represents an API Gateway VpcLink
+ */
+export interface IVpcLink extends IResource {
+  /**
+   * Physical ID of the VpcLink resource
+   * @attribute
+   */
+  readonly vpcLinkId: string;
+}
+
+/**
+ * Properties for a VpcLink
+ */
+export interface VpcLinkProps {
+  /**
+   * The VPC in which the private resources reside.
+   */
+  readonly vpc: ec2.IVpc;
+
+  /**
+   * The name used to label and identify the VPC link.
+   * @default - automatically generated name
+   */
+  readonly vpcLinkName?: string;
+
+  /**
+   * A list of security groups for the VPC link.
+   *
+   * @default - no security groups. Use `addSecurityGroups` to add security groups
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+}
+
+
+/**
+ * Define a new VPC Link
+ * Specifies an API Gateway VPC link for a HTTP API to access resources in an Amazon Virtual Private Cloud (VPC).
+ */
+export class VpcLink extends Resource implements IVpcLink {
+  /**
+   * Import a VPC Link by its Id
+   */
+  public static fromVpcLinkId(scope: Construct, id: string, vpcLinkId: string): IVpcLink {
+    class Import extends Resource implements IVpcLink {
+      public vpcLinkId = vpcLinkId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Physical ID of the VpcLink resource
+   * @attribute
+   */
+  public readonly vpcLinkId: string;
+
+  private readonly subnets = new Array<ec2.ISubnet>();
+  private readonly securityGroups = new Array<ec2.ISecurityGroup>();
+
+  constructor(scope: Construct, id: string, props: VpcLinkProps) {
+    super(scope, id);
+
+    const cfnResource = new CfnVpcLink(this, 'Resource', {
+      name: props.vpcLinkName || Lazy.stringValue({ produce: () => this.node.uniqueId }),
+      subnetIds: Lazy.listValue({ produce: () => this.renderSubnets() }),
+      securityGroupIds: Lazy.listValue({ produce: () => this.renderSecurityGroups() }),
+    });
+
+    this.vpcLinkId = cfnResource.ref;
+
+    this.addSubnets(...props.vpc.privateSubnets);
+
+    if (props.securityGroups) {
+      this.addSecurityGroups(...props.securityGroups);
+    }
+  }
+
+  /**
+   * Adds the provided subnets to the vpc link
+   *
+   * @param subnets
+   */
+  public addSubnets(...subnets: ec2.ISubnet[]) {
```

Comment:
Use [`SubnetSelection`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.SubnetSelection.html) instead of `ISubnet`?

Reply:
The `vpc` itself only has `ISubnet` array as the member variable and I couldn't find an easy way to convert one to other. Hence kept it `ISubnet`